injected/src/features/password-import.js

```diff
@@ -0,0 +1,233 @@
+
+        const findWithLabel = () => {
+            return document.querySelector(this.exportButtonLabelTextSelector)
+        }
+
+        return await withExponentialBackoff(() => findInContainer() ?? findWithLabel())
+    }
+
+    /**
+     * @returns {Promise<HTMLElement|Element|null>}
+     */
+    async findSettingsElement () {
+        const fn = () => {
+            const settingsButton = document.querySelector(this.settingsButtonSelector)
+            return settingsButton
+        }
+        return await withExponentialBackoff(fn)
+    }
+
+    /**
+     * @returns {Promise<HTMLElement|Element|null>}
+     */
+    async findSignInButton () {
+        return await withExponentialBackoff(() => document.querySelector(this.signinButtonSelector))
+    }
+
+    /**
+     * Checks if the path is supported and animates/taps the element if it is.
+     * @param {string} path
+     */
+    async handleElementForPath (path) {
+        // FIXME: we need to check if the path is supported, otherwise
+        // for some reason google doesn't wait to proceed with the signin step.
+        // Not too sure why this is happening, there are no errors on the console.
+
+        if ([this.#exportButtonSettings.path, this.#settingsButtonSettings.path, this.#signInButtonSettings.path].indexOf(path) !== -1) {
+            try {
+                const { element, style, shouldTap } = await this.getElementAndStyleFromPath(path) ?? {}
+                if (element != null) {
+                    shouldTap ? this.autotapElement(element) : this.animateElement(element, style)
+                }
+            } catch {
+                console.error('password-import: handleElementForPath failed for path:', path)
+            }
+        }
+    }
+
+    /**
+     * @returns {string}
+     */
+    get exportButtonContainerSelector () {
+        return this.#exportButtonSettings?.selectors?.join(',')
+    }
+
+    /**
+     * @returns {string}
+     */
+    get exportButtonLabelTextSelector () {
+        return this.#exportButtonSettings?.labelTexts
+            .map(text => `button[aria-label="${text}"]`)
+            .join(',')
+    }
+
+    /**
+     * @returns {string}
+     */
+    get signinLabelTextSelector () {
+        return this.#signInButtonSettings?.labelTexts
+            .map(text => `a[aria-label="${text}"]:not([target="_top"])`)
+            .join(',')
+    }
+
+    /**
+     * @returns {string}
+     */
+    get signinButtonSelector () {
+        return `${this.#signInButtonSettings?.selectors?.join(',')}, ${this.signinLabelTextSelector}`
+    }
+
+    /**
+     * @returns {string}
+     */
+    get settingsLabelTextSelector () {
+        return this.#settingsButtonSettings?.labelTexts
+            .map(text => `a[aria-label="${text}"]`)
+            .join(',')
+    }
+
+    /**
+     * @returns {string}
+     */
+    get settingsButtonSelector () {
+        return `${this.#settingsButtonSettings?.selectors?.join(',')}, ${this.settingsLabelTextSelector}`
```

Comment:
Should we have defaults in case they're not specified in the config?

Reply:
We were just talking about this to craig that the android app could have a default object, for some reason if remote config is not available. 

On the JS side we could also fallback to something, not sure if it's necessary 🤔

Reply:
Probably fine given we're only injecting this in one place. We don't really have much confidence of data passed in though so one for the future; happy to leave as is :)